### PR TITLE
Accessibility improvements

### PR DESCRIPTION
**Pull request by aomarks**
Live at https://polymerlabs.github.io/playground-elements/

- Don't immediately enter edit mode when focusing the editor with Tab, because otherwise we're a Tab trap, because Tabs insert literal tabs. Instead, show a prompt to "Press Enter to start editing". Clicking into editable region with mouse works like before. Pressing Escape re-focuses the editor, allowing Tab to be used to focus next again.

- Switch to CodeMirror's `contenteditable` input style. This makes it behave like a normal content editable region, so it's much better for screen readers (see https://codemirror.net/doc/manual.html#option_inputStyle -- it's the new default in CodeMirror 6 for this reason).

- Mark line numbers with `aria-hidden` so that they don't get read aloud by screen readers (see https://github.com/codemirror/CodeMirror/issues/6578).

### CODE REVIEW

**Comment by arthurevans** [comment]
I'm seeing a really weird issue on this. When I hit space in the editor, it's taking me to the beginning of the line.

**Comment by aomarks** [comment]
> I'm seeing a really weird issue on this. When I hit space in the editor, it's taking me to the beginning of the line.
> Edit: actually, it appears to be any character. The second character I type always jumps the cursor to the start of the current line.

This turned out to be a CodeMirror + Shadow DOM compatibility bug. I've fixed it upstream in https://github.com/codemirror/CodeMirror/pull/6651, which is already merged. Since we generate our own CodeMirror build, for now I've just updated our `devDependency` to this git SHA so that we can use the fix right away.

**Comment by aomarks** [comment]
Also got the `aria-hidden=true` fix merged upstream: https://github.com/codemirror/CodeMirror/pull/6652

**Review by sorvell — APPROVED** [review]

**Review by justinfagnani — APPROVED** [review]

File: src/playground-code-editor.ts [hunk 1]
```diff
@@ -49,12 +51,40 @@ export class PlaygroundCodeEditor extends LitElement {
...
+        position: relative;
+      }
+
       .CodeMirror {
         height: 100% !important;
         font-family: inherit !important;
         border-radius: inherit;
       }
 
+      #keyboardHelpScrim {
```
[the remark this answers is not in the document]
Comment (aomarks) [reply src/playground-code-editor.ts:65]: Done in https://github.com/PolymerLabs/playground-elements/pull/127

File: src/playground-code-editor.ts [hunk 2]
```diff
@@ -49,12 +51,40 @@ export class PlaygroundCodeEditor extends LitElement {
...
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        background: transparent;
+        z-index: 999;
+        pointer-events: none;
+      }
+
+      #keyboardHelp {
+        background: #00000099;
```
[the remark this answers is not in the document]
Comment (aomarks) [reply src/playground-code-editor.ts:80]: https://github.com/PolymerLabs/playground-elements/pull/127

File: src/playground-code-editor.ts [hunk 3]
```diff
@@ -212,15 +251,34 @@ export class PlaygroundCodeEditor extends LitElement {
...
+        @focus=${this._onFocus}
+        @blur=${this._onBlur}
+        @keydown=${this._onKeyDown}
       >
-        <div part="diagnostic-tooltip">
-          ${this._tooltipDiagnostic?.diagnostic.message}
+        ${this._showKeyboardHelp
+          ? html`<div id="keyboardHelpScrim">
+              <p id="keyboardHelp">
+                Press Enter to start editing<br />Press Escape to exit editor
```
[the remark this answers is not in the document]
Comment (aomarks) [reply src/playground-code-editor.ts:269]: Done in https://github.com/PolymerLabs/playground-elements/pull/127

**Pull request merged**